dsgrid/dataset/dataset_schema_handler_one_table.py

```diff
@@ -96,3 +98,7 @@ def get_unique_dimension_rows(self):
         self._check_null_value_in_unique_dimension_rows(dim_table)
 
         return dim_table
+
+    @track_timing(timer_stats_collector)
+    def filter_data(self, dimensions: List[DatasetSimpleModel]):
+        assert False, "not supported yet"
```

Comment:
This would be more or less similar to the two-table. Based on your code:
```
    @track_timing(timer_stats_collector)
     def filter_data(self, dimensions: List[DatasetSimpleModel]):
         load_data = self._load_data
         pivot_dimension_type = self.get_pivot_dimension_type()
         pivoted_columns = set(self.get_pivot_dimension_columns())
         pivoted_columns_to_keep = set()
         load_data_columns = set(load_data.columns)
         for dim in dimensions:
             column = dim.dimension_type.value
             if column in load_data_columns:
                 load_data = load_data.filter(load_data[column].isin(dim.record_ids))
             elif dim.dimension_type == pivot_dimension_type:
                 pivoted_columns_to_keep.update(set(dim.record_ids))
             # else trivial dimension

         # Bring it all into memory so that we can delete the original file.
         load_data2 = load_data.coalesce(1).cache()
         load_data2.count()
         overwrite_dataframe_file(self._config.load_data_lookup_path, lookup2)
         logger.info("Rewrote simplified %s", self._config.load_data_lookup_path)
         ids = next(iter(lookup2.select("id").distinct().select(F.collect_list("id")).first()))

         load_df = self._load_data.filter(self._load_data.id.isin(ids))
         pivoted_columns_to_remove = list(pivoted_columns.difference(pivoted_columns_to_keep))
         load_df = load_df.drop(*pivoted_columns_to_remove)
         overwrite_dataframe_file(self._config.load_data_path, load_df)
         logger.info("Rewrote simplified %s", self._config.load_data_path)
```

Reply:
Yeah, I know it should be straightforward. I didn't want to add code that I couldn't test. Let's add this as a separate effort when the data can be tested.